lms/djangoapps/instructor_task/tests/test_tasks_helper.py

```diff
@@ -202,3 +203,43 @@ def test_report_stores_results(self, mock_time):
 
                     self.assertEqual(return_val, UPDATE_STATUS_SUCCEEDED)
                     mock_store_rows.assert_called_once_with(self.course.id, filename, [test_header] + test_rows)
+
+
+class TestInstructorCourseForumsReport(TestReportMixin, InstructorTaskCourseTestCase):
+    """
+    Tests that course forums usage report generation works.
+    """
+    def setUp(self):
+        self.course = CourseFactory.create(org=TEST_COURSE_ORG,
+                                           number=TEST_COURSE_NUMBER,
+                                           display_name=TEST_COURSE_NAME)
+
+        self.current_task = Mock()
+        self.current_task.update_state = Mock()
+
+    def test_report_fails_if_error(self):
+        with patch('instructor_task.tasks_helper.collect_course_forums_data') as mock_collect_data:
+            mock_collect_data.side_effect = KeyError
+
+            with patch('instructor_task.tasks_helper._get_current_task') as mock_current_task:
+                mock_current_task.return_value = self.current_task
+
+                self.assertEqual(push_course_forums_data_to_s3(None, None, self.course.id, None, 'generated'), UPDATE_STATUS_FAILED)
+
+    @patch('instructor_task.tasks_helper.datetime')
+    def test_report_stores_results(self, mock_time):
+        start_time = datetime.now(UTC)
+        mock_time.now.return_value = start_time
+
+        test_header = ['Date', 'Type', 'Number', 'Up Votes', 'Down Votes', 'Net Points']
+        test_rows = [['row1_field1', 'row1_field2'], ['row2_field1', 'row2_field2']]
+
+        with patch('instructor_task.tasks_helper._get_current_task') as mock_current_task:
+            mock_current_task.return_value = self.current_task
+
+            with patch('instructor_task.tasks_helper.collect_ora2_data') as mock_collect_data:
+                mock_collect_data.return_value = (test_header, test_rows)
```

Comment:
I think I remember you saying its difficult to actually test if collect_course_forums_data actually works? That's kind of unfortunate, cuz its a nontrivial chunk of code, but I guess the worst that can happen is inaccurate forums data